### ISSUE TITLE
All performance metrics are consistently returning as 0

### ISSUE THREAD

**Issue by SkidroRLD** [ultralytics/yolov5#11601]
### Search before asking

- [ ] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

So while training a YoloV8 on a custom dataset all of my performance metrics are returned as 0. Also the confusion matrix is also slightly confusing for this. I saw another similar post however even when I tried running on CPU instead of GPU the error persisted.
I am using an Nvidia RTX 3080ti for my training. Below are attatched screenshots for the same:
![image](https://github.com/ultralytics/yolov5/assets/56610195/6405d77a-fd2
![confusion_matrix_normalized](https://github.com/ultralytics/yolov5/assets/56610195/248da26c-cc8f-4307-965f-8b98dde1987a)
e-47e6-93d2-6438cc919bf8)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@SkidroRLD hello, thank you for reaching out regarding this issue! In order to better assist you, could you please provide additional information such as your YOLOv5 version, command line arguments used for training, and any logs or error messages that were produced during the training process?

From the screenshots, it appears that the confusion matrix is not showing any true positives, which might suggest an issue with the training dataset or annotations. Additionally, training performance metrics returning as 0 could indicate an issue with the training process itself.

Please provide more information so we can better assist you in resolving this issue. Thank you!

**Comment by SkidroRLD** [comment]
Sure! Im using Yolov5 Medium, training through a python script I wrote but the code is 
model.train(data='wildfire.yaml', epochs=20, val = True)
Alll the annotations were made using LabelImg for the images. The yaml looks like this:
![image](https://github.com/ultralytics/yolov5/assets/56610195/b5dff7a7-da5f-4af7-a21f-ab9dc7ddfec2)
Regarding the confusion matrix, I do get this error as well:
 ConfusionMatrix plot failure: Invalid format specifier

**Comment by glenn-jocher** [comment]
Hi @SkidroRLD, thank you for providing more information on your issue with the performance metrics showing as 0. 

Regarding the confusion matrix error you mentioned, it seems like there might be an issue with the format specifier. Could you please provide more information on what format specifier you are using and where it is located in your code?

Regarding the performance metrics issue, could you try adjusting the training parameters such as the batch size or learning rate to see if there is any improvement? Additionally, you might want to check if there are any issues with your dataset or annotations, as it appears that the confusion matrix is not showing any true positives.

Please let us know if any of these suggestions help resolve your issue, and feel free to provide additional information or code snippets if needed. Thanks!

**Comment by SkidroRLD** [comment]
I don't know honestly, what a format specifier is and I'm sure I havent messed with any such variable/property in my code. Also, what issues should I be trying to look for in my dataset?

**Comment by glenn-jocher** [comment]
Hi @SkidroRLD,

A format specifier is a special character used in formatting strings, such as the `%` character that is used with the `print()` function. It seems that the error you're seeing might be related to how the confusion matrix is being plotted or displayed. Could you please provide more information on where the error is occurring and which library or code is generating it? 

As for issues with your dataset, there could be a few things to check for. One thing to consider is the quality of the annotations. You might want to verify that the annotations are accurate and correct, and that they are properly formatted in the format that YOLOv5 expects. Additionally, you could try checking for class imbalances in your dataset, as having too few samples for a particular class might make it difficult for the model to learn and recognize objects in that class.

I hope this information helps! Please let us know if you have any further questions or if you need additional assistance.

**Comment by SkidroRLD** [comment]
![image](https://github.com/ultralytics/yolov5/assets/56610195/2d6ef712-a185-4b5b-91e4-c4702b9907e1)
I wish I could provide more information as to where the format specifier is being used to be honest. It pops up after every end of training point and end of validation point. I ran it with the base Coco dataset and these problems d
on't occur, leading me ot believe this may be a problem in my dataset:
![DJI_20220712132453_0244](https://github.com/ultralytics/yolov5/assets/56610195/f9c8344f-e249-427a-a834-66840bf7a5bf)
[DJI_20220712132453_0244.txt](https://github.com/ultralytics/yolov5/files/11580595/DJI_20220712132453_0244.txt)
This is an example of the annotations I have done, I hope this can provide more insight

**Comment by glenn-jocher** [comment]
@SkidroRLD hi there,

Thank you for providing more information on your YOLOv5 issue. Based on the information provided, it seems like there might be an issue with the annotations in your dataset that is causing the performance metrics to show as 0 and the confusion matrix to not show any true positives.

To investigate further, we recommend verifying the accuracy and correctness of the annotations and ensuring that they are formatted correctly in the format that YOLOv5 expects. Additionally, checking for class imbalances within the dataset might also be helpful.

Please let us know if any of these suggestions help resolve your issue, or if you need any further support or assistance.

**Comment by SkidroRLD** [comment]
Well, I have more news! I was able to sort it out somehow for the performance metrics I think. But now, I'm faced with a different error where except for one class all my other class metrics are 0? Is this perhaps because of dataset imbalance?
![image](https://github.com/ultralytics/yolov5/assets/56610195/ac319e77-ed83-404a-8438-491d5cf280ef)

**Comment by glenn-jocher** [comment]
@SkidroRLD hello! 

Thanks for reaching out to us about your issue with YOLOv5 and the performance metrics returning as zeros for all classes except one. 

This issue may indeed be due to dataset imbalance, where one class has significantly more samples than the others. This can cause the model to overfit to the dominant class and result in poor performance on the other classes. 

To address this issue, you could try to balance the dataset by either adding more samples to the under-represented classes or removing samples from the over-represented classes. Another possible solution is to use weighted loss functions to give more importance to the under-represented classes during training. 

Please let us know if any of these solutions help resolve your issue, or if you need any further support or assistance.
